apis/python/src/tiledbsoma/indexed_dataframe.py

```diff
@@ -200,18 +201,46 @@ def read(
                 query_condition=query_condition,
             )
 
+            # Acceptable ways to index:
+            # * None
+            # * A sequence of coordinates is accepted, one per dimension.
+            # * Sequence length must be at least one and <= number of dimensions.
+            # * If the sequence contains missing coordinates (length less than number of dimensions),
+            #   then "slice(None)" is assumed for the missing dimensions.
+            # * Per-dimension, explicitly specified coordinates can be one of: None, a value, a
+            #   list/ndarray/paarray/etc of values, a slice, etc.
+
             if ids is not None:
-                dim_name = A.schema.domain.dim(0).name
-                if isinstance(ids, list):
-                    sr.set_dim_points(dim_name, ids)
-                elif isinstance(ids, pa.ChunkedArray):
-                    sr.set_dim_points(dim_name, ids)
-                elif isinstance(ids, pa.Array):
-                    sr.set_dim_points(dim_name, pa.chunked_array(ids))
-                elif isinstance(ids, slice):
-                    lo_hi = util.slice_to_range(ids)
-                    if lo_hi is not None:
-                        sr.set_dim_ranges(dim_name, lo_hi)
+                if not (isinstance(ids, list) or isinstance(ids, tuple)):
```

Comment:
the comment above says `ids` can be any Sequence, but this raises an error if not a list or tuple.  There are many other useful sequences (ie. range, various iterators, etc) - I think we should support them.

you can check type by doing `isinstance(ids, collections.abc.Sequence)`